src/java/org/apache/cassandra/service/disk/usage/DiskUsageMonitor.java

```diff
@@ -79,14 +81,28 @@ public void start(Consumer<DiskUsageState> notifier)
     {
         // start the scheduler regardless guardrail is enabled, so we can enable it later without a restart
         ScheduledExecutors.scheduledTasks.scheduleAtFixedRate(() -> {
-
-            if (!Guardrails.localDataDiskUsage.enabled(null))
+            boolean currentEnabledStatus = Guardrails.localDataDiskUsage.enabled(null);
+            boolean oldEnabledStatus = enabled;
+            enabled = currentEnabledStatus;
+            if (!currentEnabledStatus)
```

Comment:
this `if` is ever entered only in case `currentEnabledStatus` is `false`.

Then you do 

    if (currentEnabledStatus != oldEnabledStatus)

That means that `currentEnabledStatus` is `false`, which can't be equal to `oldEnbledStatus`.

That means that in order to call `onDiskUsageGuardrailDisabled` method, `oldEnabledStatus` has to be `true`.

This can then be simplified to

    if (!currentEnabledStatus)
    {
        if (oldEnabledStatus)
            onDiskUsageGuardrailDisabled(notifier);
    }
    else
        updateLocalState(getDiskUsage(), notifier);

Which can be further simplified to

    if (!currentEnabledStatus && oldEnabledStatus)
        onDiskUsageGuardrailDisabled(notifier);
    else
        updateLocalState(getDiskUsage(), notifier);